stdlib/public/runtime/Casting.cpp

```diff
@@ -349,42 +346,38 @@ TwoWordPair<const char *, uintptr_t>::Return
 swift_getTypeName(const Metadata *type, bool qualified) {
   using Pair = TwoWordPair<const char *, uintptr_t>;
   using Key = llvm::PointerIntPair<const Metadata *, 1, bool>;
-  
-  static pthread_rwlock_t TypeNameCacheLock = PTHREAD_RWLOCK_INITIALIZER;
+
+  static ReadWriteLock TypeNameCacheLock;
```

Comment:
Decided to use `static Lazy<ReadWriteLock> TypeNameCacheLock;` for this situation. ..?